html/includes/common/top-devices.inc.php

```diff
@@ -0,0 +1,125 @@
+<?php
+/* Copyright (C) 2015 Sergiusz Paprzycki <serek@walcz.net>
+ * 
+ * This widget is based on legacy frontpage module created by Paul Gear.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * 
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.See the
+ * GNU General Public License for more details.
+ * 
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>. */
+
+/**
+ * Top devices by traffic
+ * @author Sergiusz Paprzycki
+ * @copyright 2015 Sergiusz Paprzycki <serek@walcz.net>
+ * @license GPL
+ * @package LibreNMS
+ * @subpackage Widgets
+ */
+
+if( defined('show_settings') || empty($widget_settings) ) {
+    $common_output[] = '
+<form class="form" onsubmit="widget_settings(this); return false;">
+  <div class="form-group">
+    <div class="col-sm-6">
+      <label for="graph_type" class="control-label">Number of Devices: </label>
+    </div>
+    <div class="col-sm-4">
+      <input class="form-control" name="device_count" id="input_'.$unique_id.'" placeholder="ie. 5" value="'.$widget_settings['device_count'].'">
+    </div>
+  </div>
+  <div class="clearfix"></div>
+  <div class="form-group">
+    <div class="col-sm-6">
+      <label for="graph_type" class="control-label">Time interval (minutes): </label>
+    </div>
+    <div class="col-sm-4">
+      <input class="form-control" name="time_interval" id="input_'.$unique_id.'" placeholder="ie. 15" value="'.$widget_settings['time_interval'].'">
```

Comment:
Ooops.. sorry! That's fixed now in my local repo, but will defer PR until SQL is sorted - see below.